src/dune_rules/scope.ml

```diff
@@ -154,13 +154,15 @@ module DB = struct
       Lib.DB.Resolve_result.redirect (Some scope.db) (Loc.none, name)
     | Some (Name name) -> Lib.DB.Resolve_result.redirect None name
 
-  let public_theories ~find_db coq_stanzas =
+  let public_theories ~find_db coq_stanzas () =
     List.filter_map coq_stanzas ~f:(fun (dir, (stanza : Coq_stanza.Theory.t)) ->
         if Option.is_some stanza.package then
           Some (stanza, Coq_lib.DB.Theory dir)
         else None)
     |> Coq_lib.DB.create_from_coqlib_stanzas ~find_db ~parent:None
 
+  let public_theories ~find_db coq_stanzas = Lazy.from_fun (public_theories ~find_db coq_stanzas)
```

Comment:
Isn't it more efficient to just use `lazy` above?

Reply:
No